### ISSUE TITLE
distributor blocking on ingestion

### ISSUE THREAD

**Issue by woodsaj** [grafana/loki#300]
I am seeing really high avg distributor latency, while p99 is acceptable.  This will be due to a small number of requests getting blocked and taking hours to complete.

I took at goroutine dump from a distributor and there are ~500 goroutines that are stuck at 

https://github.com/grafana/loki/blob/master/pkg/distributor/distributor.go#L171-L176

```
--
goroutine 20622 [select, 617 minutes]:
github.com/grafana/loki/pkg/distributor.(*Distributor).Push(0xc0000d4cd0, 0x1401d00, 0xc000ca46f0, 0xc000304820, 0x13fa700, 0xc000304820, 0x2)
        /go/src/github.com/grafana/loki/pkg/distributor/distributor.go:171 +0xac8
github.com/grafana/loki/pkg/distributor.(*Distributor).PushHandler(0xc0000d4cd0, 0x1401340, 0xc000304780, 0xc000425700)
--
goroutine 52981 [select, 383 minutes]:
github.com/grafana/loki/pkg/distributor.(*Distributor).Push(0xc0000d4cd0, 0x1401d00, 0xc000df8c00, 0xc0028239c0, 0x13fa700, 0xc0028239c0, 0x2)
        /go/src/github.com/grafana/loki/pkg/distributor/distributor.go:171 +0xac8
github.com/grafana/loki/pkg/distributor.(*Distributor).PushHandler(0xc0000d4cd0, 0x1401340, 0xc0028239a0, 0xc00219fa00)
--
```

**Comment by tomwilkie** [comment]
Thanks @woodsaj, looking into this.

**Comment by tomwilkie** [comment]
Fixed by #301
